src/components/shared/Property.tsx

```diff
@@ -46,7 +46,7 @@ function isExternalRefSchemaNode(schemaNode: SchemaNode) {
 export const Property: React.FunctionComponent<IProperty> = ({ node: treeNode, onGoToRef }) => {
   const { path, schemaNode: node } = getSchemaNodeMetadata(treeNode);
   const type = isRefNode(node) ? '$ref' : isCombinerNode(node) ? node.combiner : node.type;
-  const subtype = isArrayNodeWithItems(node) ? inferType(node.items) : void 0;
+  const subtype = isArrayNodeWithItems(node) ? (hasRefItems(node) ? '$ref' : inferType(node.items)) : void 0;
```

Comment:
Let's just included that check within `inferType` method

Reply:
Actually that may be not so perfect. `inferType` is used in other places and putting `$ref` there might require to update those parts.
For instance `processNode` in `walk` function uses type that is falsy. We can of course change it to make use of $ref` type but not sure if it's worth it.

Reply:
> Actually that may be not so perfect. inferType is used in other places and putting $ref there might require to update those parts.


Yeah, I'm aware.

> For instance processNode in walk function uses type that is falsy. We can of course change it to make use of $ref` type but not sure if it's worth it.

well, in fact it might require a chunk of work to get it done, so let's leave it as is for now.